armory/scenarios/poison.py

```diff
@@ -386,6 +390,46 @@ def finalize_results(self):
             log.info(
                 f"Test targeted misclassification accuracy: {self.poisoned_targeted_test_metric.mean():.2%}"
             )
+        
+        self.results["N_poisoned_data"] = int(len(self.poison_index))
+
+        if self.use_filtering_defense:
+           
+            filtered = (1 - self.indices_to_keep).astype(np.bool)
+            poisoned = np.zeros_like(self.y_clean).astype(np.bool)
+            poisoned[self.poison_index] = True
+
+            false_negatives = int(np.sum(~filtered & poisoned))
+            true_positives = int(np.sum(filtered & poisoned))
+            true_negatives = int(np.sum(~filtered & ~poisoned))
+            false_positives = int(np.sum(filtered & ~poisoned))
+            
+            false_negative_rate = false_negatives / np.sum(poisoned)
+            true_positive_rate = true_positives / np.sum(poisoned)
+            true_negative_rate = true_negatives / np.sum(~poisoned)
+            false_positive_rate = false_positives / np.sum(~poisoned)
+            
+            f1_score = true_positives / (true_positives + .5 * (false_positives + false_negatives))
+            
+            self.results["filter_true_positives"] = true_positives
+            self.results["filter_false_positives"] = false_positives
+            self.results["filter_true_negatives"] = true_negatives
+            self.results["filter_false_negatives"] = false_negatives
+            self.results["filter_true_positive_rate"] = true_positive_rate
+            self.results["filter_false_positive_rate"] = false_positive_rate
+            self.results["filter_true_negative_rate"] = true_negative_rate
+            self.results["filter_false_negative_rate"] = false_negative_rate
+            self.results["filter_f1_score"] = f1_score
+
+            self.results["fraction_data_removed_by_filter"] = filtered.mean()
+            self.results["N_datapoints_removed_by_filter"] = int(filtered.sum())
+            
+
+        for y in np.unique(self.y_clean):
+            self.results[f"N_samples_in_class_{y}"] = int(np.sum(self.y_clean == y))
```

Comment:
for `"N_samples_in_class_y"`, I was thinking instead of `int(np.sum(self.y_clean == y))`, you could just do `len(self.class_accuracy_dict[y])`. 

However when I `breakpoint()` in the code to confirm this, the results aren't what I expect:

```
(Pdb) np.sum(self.y_clean == 0)
210

(Pdb) len(self.class_accuracy_dict[0])
60
```
After looking closer, I think it's because the former comes from the train set while the latter is from the test set? If this is correct, I think it'd be clearer to add `train` and `test` to more variable names and dict keys

Side note: when I was looking at the number of data points in `self.class_accuracy_dict` I noticed it was 12630:

```
>>> num_per_class = [len(self.class_accuracy_dict[k]) for k in self.class_accuracy_dict]
>>> sum(num_per_class)
12630
```

This isn't divisible by the batch_size (512), so is this just the total length of the test set and the last batch has fewer examples? Otherwise I'm not sure how this number is derived

Reply:
You are right about train/test data.  I'm clarifying the variable names.

You also seem to be right about the last batch having fewer samples.  When I print out the batch sizes, I'm seeing 24 batches of size 512 and one of size 342 for a total of 12630